### PR TITLE
#118 Add highlights support

### PR DESCRIPTION
**Pull request by gnuzzz**

### CODE REVIEW

File: java/src/main/kotlin/org/taymyr/lagom/elasticsearch/search/dsl/query/highlight/Highlight.kt [hunk 1]
```diff
@@ -0,0 +1,391 @@
...
+import java.util.Locale
+
+/**
+ * See [Elasticsearch Docs](https://www.elastic.co/guide/en/elasticsearch/reference/current/highlighting.html).
+ */
+enum class BoundaryScanner {
+    @JsonProperty("chars")
+    CHARS,
+
+    @JsonProperty("CHARS")
```
Comment (ihostage) [inline java/src/main/kotlin/org/taymyr/lagom/elasticsearch/search/dsl/query/highlight/Highlight.kt:16]:
```suggestion
    @JsonProperty("sentence")
```

File: java/src/main/kotlin/org/taymyr/lagom/elasticsearch/search/dsl/query/highlight/Highlight.kt [hunk 2]
```diff
@@ -0,0 +1,391 @@
...
+    SENTENCE,
+
+    @JsonProperty("word111")
```
Comment (ihostage) [inline java/src/main/kotlin/org/taymyr/lagom/elasticsearch/search/dsl/query/highlight/Highlight.kt:19]:
```suggestion
    @JsonProperty("word")
```

File: java/src/main/kotlin/org/taymyr/lagom/elasticsearch/search/dsl/query/highlight/Highlight.kt [hunk 3]
```diff
@@ -0,0 +1,391 @@
...
+
+    @JsonProperty("plain")
+    PLAIN,
+
+    @JsonProperty("fvh")
+    FVH
+}
+
+abstract class HighlightItem constructor(
+    @JsonProperty("boundary_chars")
```
Comment (ihostage) [inline java/src/main/kotlin/org/taymyr/lagom/elasticsearch/search/dsl/query/highlight/Highlight.kt:56]: Can you try use `@JsonNaming(PropertyNamingStrategy.SnakeCaseStrategy.class)` on class level instead?

File: java/src/main/kotlin/org/taymyr/lagom/elasticsearch/search/dsl/SearchRequest.kt [hunk 4]
```diff
@@ -29,7 +30,9 @@ data class SearchRequest @JvmOverloads constructor(
     @JsonProperty("search_after")
     val searchAfter: List<Any>? = null,
     @JsonProperty("_source")
-    val source: SourceFilter<*>? = null
+    val source: SourceFilter<*>? = null,
+    @JsonProperty("highlight")
```
Comment (ihostage) [inline java/src/main/kotlin/org/taymyr/lagom/elasticsearch/search/dsl/SearchRequest.kt:34]: Can be removed as equals property name.

**Review by ihostage — CHANGES_REQUESTED** [review]
Few small suggestions 😉

File: java/src/main/kotlin/org/taymyr/lagom/elasticsearch/search/dsl/query/highlight/Highlight.kt (hunk 1 above)
Comment (gnuzzz) [reply java/src/main/kotlin/org/taymyr/lagom/elasticsearch/search/dsl/query/highlight/Highlight.kt:16]: done

File: java/src/main/kotlin/org/taymyr/lagom/elasticsearch/search/dsl/query/highlight/Highlight.kt (hunk 2 above)
Comment (gnuzzz) [reply java/src/main/kotlin/org/taymyr/lagom/elasticsearch/search/dsl/query/highlight/Highlight.kt:19]: done

File: java/src/main/kotlin/org/taymyr/lagom/elasticsearch/search/dsl/query/highlight/Highlight.kt (hunk 1 above)
Comment (gnuzzz) [reply java/src/main/kotlin/org/taymyr/lagom/elasticsearch/search/dsl/query/highlight/Highlight.kt]: done

File: java/src/main/kotlin/org/taymyr/lagom/elasticsearch/search/dsl/query/highlight/Highlight.kt (hunk 2 above)
Comment (gnuzzz) [reply java/src/main/kotlin/org/taymyr/lagom/elasticsearch/search/dsl/query/highlight/Highlight.kt]: done

File: java/src/main/kotlin/org/taymyr/lagom/elasticsearch/search/dsl/query/highlight/Highlight.kt (hunk 3 above)
Comment (gnuzzz) [reply java/src/main/kotlin/org/taymyr/lagom/elasticsearch/search/dsl/query/highlight/Highlight.kt]: checked, it works

File: java/src/main/kotlin/org/taymyr/lagom/elasticsearch/search/dsl/SearchRequest.kt (hunk 4 above)
Comment (gnuzzz) [reply java/src/main/kotlin/org/taymyr/lagom/elasticsearch/search/dsl/SearchRequest.kt]: done

**Review by ihostage — APPROVED** [review]
LGTM

**Pull request merged**